### ISSUE TITLE
Abort shell recorder on failed mount

### ISSUE THREAD

**Issue by kodebach** [ElektraInitiative/libelektra#4826]
From looking at #4758 I had the idea, that we should abort our shell recorder tests, if the initial `kdb mount` command fails.

The `# Backup-and-Restore:` meta-command might already do that, but it isn't used everywhere, since it doesn't support things like `--with-recommends`.

It might even make sense to always abort on failed commands. But especially a failed `kdb mount` is a huge problem. It the rest of the test script might destroy the KDB and then all following tests using the KDB will also fail.

**Comment by markus2330** [comment]
Thank you for reporting! Would be interesting if `# Backup-and-Restore:` already does it.

**Comment by kodebach** [comment]
Turns out the `shell_recorder` in general does a full backup of the entire KDB before each test (maybe not a good idea for performance):

https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/shell_recorder.sh#L291

It's used for the "did the test cleanup properly" check:

https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/shell_recorder.sh#L326

But in the cleanup method at the end

https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/shell_recorder.sh#L9-L12

we just delete the backup instead of restoring it. That might be useful for local diagnosis of test failures. But in the CI it would make more sense, to just restore the backup, so the next test isn't affected.

So the first thing I would do is add an env-var (which we will set in the CI config): `ELEKTRA_SR_RESTORE` if set the `shell_recorder` restores the KDB after each test.

Regarding `#Backup-and-Restore`: I'm not 100% certain I got it correctly, but here is what I found:

It is translated into `Mountpoint:` for the `.esr` syntax by `markdown_shell_recorder`:
https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/tutorial_wrapper/markdown_shell_recorder.sh#L50-L56

> **Note**: As you can see, if there is no `#Backup-and-Restore`, we use the parent `/tests` for the following steps.

That is read by `shell_recorder` itself here:
https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/shell_recorder.sh#L182-L185

Which seems to define the `$Mountpoint` variable for all following commands and also does a `kdb export` before the first command:

https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/shell_recorder.sh#L22-L29

After the whole script, a `kdb import` is used to restore the exported config:

https://github.com/ElektraInitiative/libelektra/blob/55c609a503e1ffdabb165a23777362b057284bdd/tests/shell/shell_recorder/shell_recorder.sh#L296-L297

So in summary, there are some attempts at restoring the backup. However, `Backup-and-Restore` itself doesn't actually execute a `kdb mount`, so the original issue still stands. A failed `kdb mount` could lead to corruption by the rest of the script and that could also affect `kdb import`.
